### PR TITLE
bitfield: get first set bit

### PR DESCRIPTION
**Pull request by OlegHahm**
+ unittests

### CODE REVIEW

File: tests/unittests/tests-bitfield/tests-bitfield.c [hunk 1]
```diff
@@ -13,10 +13,12 @@
 
 #include "bitfield.h"
 
+#define BF_SIZE (5)
+
 static void test_bf_get_unset_empty(void)
 {
     int res = 0;
-    uint8_t field[5];
+    uint8_t field[BF_SIZE];
```
Comment (authmillenon) [inline tests/unittests/tests-bitfield/tests-bitfield.c]: Why not use the BITFIELD macro with `BF_SIZE == 40` here? That would make the code below a little bit easier to read:

```C
BITFIELD(field, BF_SIZE);
```

**Comment by kaspar030** [comment]
We should mention that for bitfields with at most 32bit there is a more efficient version in bitarithm.h.

File: tests/unittests/tests-bitfield/tests-bitfield.c (hunk 1 above)
Comment (OlegHahm) [inline tests/unittests/tests-bitfield/tests-bitfield.c]: I think the rational was to not rely on the BITFIELD macro here.

**Comment by OlegHahm** [comment]
> We should mention that for bitfields with at most 32bit there is a more efficient version in bitarithm.h.

Agreed - though their feature set is a little bit different.

**Comment by OlegHahm** [comment]
But maybe we could add the missing functionality to `bitarithm`. Shouldn't be too difficult (TM).

File: sys/bitfield/bitfield.c [hunk 2]
```diff
@@ -46,3 +46,27 @@ int bf_get_unset(uint8_t field[], int size)
...
+
+int bf_get_first_set(uint8_t field[], int size)
+{
+    int result = -1;
+    int nbytes = (size + 7) / 8;
+    int i = 0;
+
+    unsigned state = disableIRQ();
+
+    /* skip full bytes */
```
Comment (cgundogan) [inline sys/bitfield/bitfield.c]: skipping words until a positive match is found and then skipping bytes might converge faster

Comment (OlegHahm) [inline sys/bitfield/bitfield.c]: We had this discussion when @kaspar030 introduced `bf_get_unset` and decided against it. I think it's mostly a question of what sizes for bitfields you expect and trading code space for performance.

Comment (Kijewski) [inline sys/bitfield/bitfield.c]: I guess memory alignment could also be a problem if you read words.

**Comment by OlegHahm** [comment]
Updated. Anyone for an ACK?

**Comment by OlegHahm** [comment]
Got rid of the need for this in #3561. Should I close this one or does anyone else see a need for it?

**Comment by gebart** [comment]
Close it, you can always reopen if a use case appears.

**Pull request closed without merging**